openedx/core/djangoapps/credit/api.py

```diff
@@ -111,6 +111,91 @@ def get_credit_requirements(course_key, namespace=None):
     ]
 
 
+def get_credit_requirement_status(course_key, username):
+    """ Retrieve the user's status for each credit requirement in the course.
+
+    Args:
+        course_key (CourseKey): The identifier for course
+        username (str): The identifier of the user
+
+    Example:
+        >>> get_credit_requirement_status("course-v1-edX-DemoX-1T2015", "john")
+
+                [
+                    {
+                        "namespace": "verification",
+                        "name": "verification",
+                        "criteria": {},
+                        "status": "satisfied",
+                    },
+                    {
+                        "namespace": "reverification",
+                        "name": "midterm",
+                        "criteria": {},
+                        "status": "Not satisfied",
+                    },
+                    {
+                        "namespace": "proctored_exam",
+                        "name": "final",
+                        "criteria": {},
+                        "status": "error",
+                    },
+                    {
+                        "namespace": "grade",
+                        "name": "grade",
+                        "criteria": {"min_grade": 0.8},
+                        "status": None,
+                    },
+                ]
+
+    Returns:
+        list of requirement statuses
+    """
+    requirements = CreditRequirement.get_course_requirements(course_key)
+    statuses = []
+    for requirement in requirements:
+        status = CreditRequirementStatus.get_status(requirement, username)
```

Comment:
You can get this data in single query CreditRequirementStatus.filter(username=username, requirement__course_key=course_key) instead of using the two functions CreditRequirement.get_course_requirements(course_key) and in loop CreditRequirementStatus.get_status(requirement, username)